MapboxCoreNavigation/RouteProgress.swift

```diff
@@ -180,7 +180,7 @@ open class RouteProgress {
         self.route = route
         self.routeOptions = options
         self.legIndex = legIndex
-        self.currentLegProgress = RouteLegProgress(leg: route.legs[legIndex], stepIndex: 0, spokenInstructionIndex: spokenInstructionIndex)
+        self.currentLegProgress = try! RouteLegProgress(leg: route.legs[legIndex], stepIndex: 0, spokenInstructionIndex: spokenInstructionIndex)
```

Comment:
Concern about explicit `try!` here.
Can we show explicitly the invariant (via adding assert, say)?

Reply:
Replaced exception with precondition.